### PR TITLE
Themable components

### PR DESCRIPTION
**Pull request by snixtho**
This pull request adds a full featured theme system.

Features:
- Define default theme options for any component or manialink
- Override themes from core or other modules
- Replace components fully with other components
- Provides default theme options such as primary and secondary colors, as well as utility colors.
- Can override theme options from the config.
- Several utilities for font and colors.
- The general theme object and some utilities are available as global variables in mainalinks.
- Add all available in-game icons defined in a class for easy access.

All existing templates has been updated to use the new theme system (main reason for so many changed files).

ManiaTemplates is also updates which includes global variables.

### CODE REVIEW

File: src/EvoSC.Common/Util/ColorUtils.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    /// <returns></returns>
+    public static double Luma(IRgb color) => Math.Round(color.R * 0.2126 + color.B * 0.7152 + color.B * 0.0722);
+    
+    /// <summary>
+    /// Get the luma of a color.
+    /// </summary>
+    /// <param name="hexColor">Hex color to calculate luma from.</param>
+    /// <returns></returns>
+    public static double Luma(string hexColor) => Luma(new Hex(hexColor).ToRgb());
+    /// <summary>
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Util/ColorUtils.cs:122]:
```suggestion
    
    /// <summary>
```

File: src/EvoSC/ApplicationSetup.cs [hunk 2]
```diff
@@ -79,6 +80,8 @@ public static void SetupPipeline(this IStartupPipeline pipeline, IEvoScBaseConfi
             .Services(AppFeature.Permissions, s => s.AddEvoScPermissions())
 
             .Services(AppFeature.Manialinks, s => s.AddEvoScManialinks())
+                
+                .Services(AppFeature.Themes, s => s.AddEvoScThemes())
```
Comment (AtomicLiquid) [inline src/EvoSC/ApplicationSetup.cs:84]: Incorrect indentation.

File: src/Modules/CurrentMapModule/CurrentMapModule.csproj [hunk 3]
```diff
@@ -22,4 +22,5 @@
       </PackageReference>
     </ItemGroup>
     
+
```
Comment (AtomicLiquid) [inline src/Modules/CurrentMapModule/CurrentMapModule.csproj:25]: Is this necessary?

File: src/EvoSC.Manialinks/Util/GameIcons.cs
[the remark this answers is not in the document]
Comment (snixtho) [reply src/EvoSC.Manialinks/Util/GameIcons.cs:1]: Don't think that is necessary.

File: src/Modules/ASayModule/Templates/Announcement.mt [hunk 4]
```diff
@@ -1,49 +1,88 @@
...
+                          Width += C_IconWidth;
+          Frame.RelativePosition_V3.X = -(Width) /2.;
+          BgQuad.Size.X = Width;
+          InnerFrame.RelativePosition_V3.X = 11.;
+          if (Width == (C_MaxWidth + C_IconWidth)) {
+              declare Real Pos = Width/2. +5;
+              Label.RelativePosition_V3.X = Pos;
+              while(True) {
+                      yield;
+                      Label.RelativePosition_V3 = <(ML::Sin(Now*0.0005)) * (Width/8), 0.>;
```
Comment (araszka) [inline src/Modules/ASayModule/Templates/Announcement.mt:81]: Another possibility would be to use `AnimMgr.Add` and a sleep timeout to alternate between positions. 
I don't know if that would have any performance benefits for the client, but maybe AnimMgr can move UI elements more efficiently than pure maniascript.

**Review by araszka — APPROVED** [review]

File: src/Modules/ASayModule/Templates/Announcement.mt (hunk 4 above)
Comment (snixtho) [reply src/Modules/ASayModule/Templates/Announcement.mt:81]: Made an issue for it: https://github.com/EvoEsports/EvoSC-sharp/issues/235

**Pull request merged**